### PR TITLE
Merge the two Cachito plugins

### PR DESCRIPTION
**Pull request by chmeliik**
CLOUDBLD-8121

The resolve_remote_source plugin now does everything that
download_remote_source used to do.

The download_remote_source plugin is removed.

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by chmeliik** [comment]
PTAL. Note that the `inject_remote_sources` method is a rough equivalent of  what used to be the `run` method of the download_remote_source plugin

**Review by rcerven — APPROVED** [review]

**Comment by hjmodi** [comment]
@chmeliik the coverage has decreased, is it intended/expected?

**Comment by chmeliik** [comment]
> @chmeliik the coverage has decreased, is it intended/expected?

Hmm, it is not. I'll check

**Comment by chmeliik** [comment]
Rebased on master and added unit test for invalid JSON for the `get_request_config_files` method (I think that was the coverage decrease)

File: atomic_reactor/utils/cachito.py [hunk 1]
```diff
@@ -225,13 +226,35 @@ def get_request_env_vars(self, rid):
...
+            parsed.
+        :raises: HTTP error raised from the underlying requests library in any
+            case of the request cannot be completed.
+        """
+        endpoint = f'{self.api_url}/api/v1/requests/{rid}/configuration-files'
+        resp = self.session.get(endpoint)
+        resp.raise_for_status()
+        try:
+            return resp.json()
+        except Exception as exc:
```
Comment (hjmodi) [inline atomic_reactor/utils/cachito.py:246]: is it possible we can qualify this exception?

**Comment by hjmodi** [comment]
LGTM, just one minor comment

File: atomic_reactor/utils/cachito.py (hunk 1 above)
Comment (chmeliik) [reply atomic_reactor/utils/cachito.py:246]: The other methods in this module also catch bare `Exception` :disappointed: 

I'm fairly certain that catching ValueError would be good enough, but I'd rather not accidentally change the behavior

**Review by mkosiarc — APPROVED** [review]

**Review by hjmodi — APPROVED** [review]

**Pull request merged**